### PR TITLE
improve examples in manual

### PR DESCRIPTION
**Pull request by ringabout**
Ref https://github.com/timotheecour/Nim/issues/64#issuecomment-751620923

### CODE REVIEW

File: doc/manual.rst [hunk 1]
```diff
@@ -7062,8 +7062,10 @@ one can import C++'s templates rather easily without the need for a pattern
 language for object types:
 
 .. code-block:: nim
+  :test: "nim cpp $1"
```
Comment (ringabout) [inline doc/manual.rst:7065]: warning as errors may be useful in manual tests

Comment (timotheecour) [reply doc/manual.rst:7065]: indeed; but this should apply globally, no reason to single out just 1 test

can followup on this in https://github.com/timotheecour/Nim/issues/64#issuecomment-751982665

**Review by timotheecour — APPROVED** [review]

File: doc/manual.rst [hunk 2]
```diff
@@ -7062,8 +7062,10 @@ one can import C++'s templates rather easily without the need for a pattern
...
+
   type
-    StdMap {.importcpp: "std::map", header: "<map>".} [K, V] = object
+    StdMap [K, V] {.importcpp: "std::map", header: "<map>".} = object
```
Comment (Clyybber) [inline doc/manual.rst]:
```suggestion
    StdMap[K, V] {.importcpp: "std::map", header: "<map>".} = object
```

File: tests/cpp/ttemplatetype.nim [hunk 3]
```diff
@@ -3,7 +3,7 @@ discard """
 """
 
 type
-  Map {.importcpp: "std::map", header: "<map>".} [T,U] = object
+  Map [T,U] {.importcpp: "std::map", header: "<map>".} = object
```
Comment (Clyybber) [inline tests/cpp/ttemplatetype.nim]:
```suggestion
  Map[T,U] {.importcpp: "std::map", header: "<map>".} = object
```

**Pull request merged**